File: software/examples/csharp/ExampleThreshold.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Humidity Bricklet

	// Callback function for humidity reached callback
	static void HumidityReachedCB(BrickletHumidity sender, int humidity)
	{
		Console.WriteLine("Humidity: " + humidity/10.0 + " %RH");
		Console.WriteLine("Recommended humidity for human comfort is 30 to 60 %RH.");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletHumidity h = new BrickletHumidity(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get threshold callbacks with a debounce time of 10 seconds (10000ms)
		h.SetDebouncePeriod(10000);

		// Register humidity reached callback to function HumidityReachedCB
		h.HumidityReachedCallback += HumidityReachedCB;

		// Configure threshold for humidity "outside of 30 to 60 %RH"
		h.SetHumidityCallbackThreshold('o', 30*10, 60*10);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/java/ExampleThreshold.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletHumidity;

public class ExampleThreshold {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your Humidity Bricklet
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletHumidity h = new BrickletHumidity(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get threshold callbacks with a debounce time of 10 seconds (10000ms)
		h.setDebouncePeriod(10000);

		// Add humidity reached listener
		h.addHumidityReachedListener(new BrickletHumidity.HumidityReachedListener() {
			public void humidityReached(int humidity) {
				System.out.println("Humidity: " + humidity/10.0 + " %RH");
				System.out.println("Recommended humidity for human comfort is 30 to 60 %RH.");
			}
		});

		// Configure threshold for humidity "outside of 30 to 60 %RH"
		h.setHumidityCallbackThreshold('o', 30*10, 60*10);

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}
